### PR TITLE
Write a readme

### PR DESCRIPTION
**Pull request by elwinar**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,145 @@
 # zConfig [![Build Status](https://travis-ci.org/synthesio/zconfig.svg?branch=master)](https://travis-ci.org/synthesio/zconfig) [![Godoc](http://img.shields.io/badge/godoc-reference-blue.svg?style=flat)](https://godoc.org/github.com/synthesio/zconfig) [![license](http://img.shields.io/badge/license-MIT-red.svg?style=flat)](https://raw.githubusercontent.com/synthesio/zconfig/master/LICENSE.md)
+
+`zConfig` is an extensible, reflection-based configuration and dependency
```
Comment (beorn-) [inline README.md]: golang reflection-based

File: README.md [hunk 2]
```diff
@@ -1 +1,145 @@
...
+their description and default values.
+
+```bash
+$ ./a.out --help
+Keys:
+addr	ADDR	address the server should bind to	(:80)
+```
+
+Configurations can be nested into structs to improve usability, and the keys of
+the final paraeters are prefixed by the keys of all parents.
```
Comment (beorn-) [inline README.md]: parameters

File: README.md [hunk 3]
```diff
@@ -1 +1,145 @@
...
+
+```bash
+$ ./a.out --help
+Keys:
+server.addr	SERVER_ADDR	address the server should bind to	(:80)
+```
+
+### Dependency Injection
+
+The zconfig processor undertand a set of tag used for injecting one field into
```
Comment (beorn-) [inline README.md]: understands

Comment (beorn-) [reply README.md]: tags

File: README.md [hunk 4]
```diff
@@ -1 +1,145 @@
...
+		Root string `inject-as:"bar-root"`
+	}
+}
+```
+
+Note that the injection system isn't tied to the configuration one: you don't
+need your injection source or target to be part of a chain of `key`ed structs.
+
+Any type can be injected as long as the source is _assignable_ to the target,
+allowing whole structs to be shared by multiple configuration parts.
```
Comment (beorn-) [inline README.md]: by with or among ?

**Comment by sayoun** [comment]
:+1:

**Pull request merged**